src/test/java/com/gavinfenton/quizolation/service/QuizServiceTest.java

```diff
@@ -144,6 +148,29 @@ public void testUpdateQuizSetsIdBeforeSaveAndReturnsFromRepository() {
         assertEquals(quizExpected, quizActual);
     }
 
+    @Test
+    public void testAddTeamToQuizCallsAndReturnsFromRepo() throws Exception {
+        //Given
+        Quiz quizSaving = new Quiz();
+        Long quizIdSavingExpected = 32L;
+        Quiz quizExpected = new Quiz();
+        Team teamSaving = new Team();
+        Long teamIdSavingExpected = 1L;
+        Team teamExpected = new Team();
+        given(teamService.getTeam(teamIdSavingExpected)).willReturn(teamExpected);
+        given(quizRepository.findById(quizIdSavingExpected)).willReturn(Optional.of(quizSaving));
+        given(quizRepository.save(quizSaving)).willReturn(quizExpected);
+
+        //When
+        Quiz quizActual = quizService.addTeamToQuiz(quizIdSavingExpected, teamIdSavingExpected);
+
+        //Then
+        verify(teamService).getTeam(teamIdSavingExpected);
+        verify(quizRepository).findById(quizIdSavingExpected);
+        verify(quizRepository).save(quizSaving);
+        assertEquals(quizExpected, quizActual);
```

Comment:
quizSaving and quizExpected could have the same hashcode value as no differences have been guaranteed by updating a field on one of them in Given

Reply:
Resolved (i think)

Reply:
Nope, quizExpected should have something set on it to make sure that an empty Quiz isn't returned